src/server/engine_shard_set.cc

```diff
@@ -861,18 +858,16 @@ void EngineShardSet::TEST_EnableCacheMode() {
 }
 
 ShardId Shard(string_view v, ShardId shard_num) {
-  bool has_hashtags = false;
   if (ClusterConfig::IsEnabledOrEmulated()) {
     string_view v_hash_tag = ClusterConfig::KeyTag(v);
     if (v_hash_tag.size() != v.size()) {
-      has_hashtags = true;
       v = v_hash_tag;
     }
   }
 
   XXH64_hash_t hash = XXH64(v.data(), v.size(), 120577240643ULL);
 
-  if (has_hashtags) {
+  if (RoundRobinSharder::IsEnabled()) {
```

Comment:
As I said above, I'd keep this `if` inside `TryGetShardId()`, and will remove it from here, so that the call to `TryGetShardId()` is unconditional. I think it's better like that.

Reply:
The reason I applied Adi's suggestion was because `TryGetShardId` is only used in one place and there is really no point to make the call if RoundRobinSharder is not enabled. I agree with you that in general, it's best for the function itself to do all the checking but for this particular case both are equally good IMO. It's a very small piece of code which I think it's not worth tweaking more

Reply:
I still think it'd be better, but your call..